backend/channels.py

```diff
@@ -663,9 +805,207 @@ async def delete_channel_muxes(config, mux_uuid):
 
 
 async def map_all_services(config):
```

Comment:
I am not sure what this is solving or modifying. At the moment it seems like a lot of work went into this code you have written, but I cannot see any benefit to it when I add and remove channels with my EPG and playlists. What was wrong with simply running "Map all services" in TVH as this will successfully map everything that it finds. The MUX and the XMLTV EPG guide has been custom generated with the same channel_number.